src/protocols/secure_channel/CheckinMessage.h

```diff
@@ -45,28 +45,39 @@ class DLL_EXPORT CheckinMessage
     /**
      * @brief Generate Check-in Message payload
      *
-     * @param key       Key with which to encrypt the check-in payload
-     * @param counter   Check-in counter
-     * @param appData   Application Data to incorporate within the Check-in message. Allowed to be empty.
-     * @param output    Buffer in Which to store the generated payload. SUFFICIENT SPACE MUST BE ALLOCATED by the caller
-     *                  Required Buffer Size is : GetCheckinPayloadSize(appData.size())
+     * @note Function requires two key handles to generate the Check-In message.
+     *       Due to PSA requirements, the same key handle cannot be used for AES-CCM and HMAC-SHA-256 operations.
+     *
+     * @param[in]  aes128KeyHandle   Key handle with which to encrypt the check-in payload with the AEAD
+     * @param[in]  hmac128KeyHandle  Key handle with which to generate the nonce the check-in payload with the HMAC
+     * @param[in]  counter           Check-in counter
+     * @param[in]  appData           Application Data to incorporate within the Check-in message. Allowed to be empty.
```

Comment:
during testing, we found it would return error if appData is empty. @thivya-amazon  @mkardous-silabs

Reply:
it seems you have not yet added empty appData test?

Reply:
appData has to be of size 6 in the client to retrieve counter(4 bytes) and activemodethreshold(2 bytes).

Reply:
@yunhanw-google a Test is already present for the Parse function with empty payload 
https://github.com/project-chip/connectedhomeip/blob/cfcde225a41c4a2f8b07be6974f1c494d75f28e6/src/protocols/secure_channel/tests/TestCheckinMsg.cpp#L185

Before this PR, activeModeThreshold was not being encoded in the check-in message. Could that be what is causing the error?

Reply:
The payload generator should be resilient to *arbitrary* app data, since that is a layer above the actual encoding in the message. However, the spec does mandate Appdata presence. I would argue for GenerateCheckingMessagePayload to be tolerant of empty appData, but tested both for empty appData and non-empty appData.

Reply:
Underlying limitation is the appData buffer is used as a work buffer so it always need to be at least the size of the counter.

Clarified the comment to make sure api consummers provides the minimum requirements.